src/extensions/replay/sessionrecording.ts

```diff
@@ -582,11 +583,10 @@ export class SessionRecording {
         this.sessionId = sessionId
 
         if (
-            returningFromIdle ||
-            ([FULL_SNAPSHOT_EVENT_TYPE, META_EVENT_TYPE].indexOf(event.type) === -1 &&
-                (windowIdChanged || sessionIdChanged || isUndefined(this._fullSnapshotTimer)))
+            [FULL_SNAPSHOT_EVENT_TYPE, META_EVENT_TYPE].indexOf(event.type) === -1 &&
+            (windowIdChanged || sessionIdChanged || isNull(this.lastSessionFullSnapshot))
```

Comment:
silly tired paul, this isn't correct... we need to check if the tracked last snapshot session is the current session